File: app/src/main/java/com/bytedance/tiktok/view/MarqueeTextView.java
package com.bytedance.tiktok.view;

import android.content.Context;
import android.util.AttributeSet;
import android.widget.TextView;

/**
 * create by libo
 * create on 2020-05-21
 * description 跑马灯textview
 */
public class MarqueeTextView extends TextView {

    public MarqueeTextView(Context context) {
        super(context);
    }

    public MarqueeTextView(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public MarqueeTextView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override
    public boolean isFocused(){
        return true;
    }

}

File: app/src/main/java/com/bytedance/tiktok/view/MarqueeTextView.kt
package com.bytedance.tiktok.view

import android.content.Context
import android.util.AttributeSet
import android.widget.TextView

/**
 * create by libo
 * create on 2020-05-21
 * description 跑马灯textview
 */
class MarqueeTextView : TextView {
    constructor(context: Context?) : super(context) {}
    constructor(context: Context?, attrs: AttributeSet?) : super(context, attrs) {}
    constructor(context: Context?, attrs: AttributeSet?, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {}

    override fun isFocused(): Boolean {
        return true
    }
}